watchtower-plugin/src/dbm.rs

```diff
@@ -216,19 +216,29 @@ impl DBM {
         &self,
         tower_id: TowerId,
         user_id: UserId,
+        subscription_start: Option<u32>,
+        subscription_expiry: Option<u32>,
     ) -> Option<RegistrationReceipt> {
-        let mut stmt = self
-            .connection
-            .prepare(
-                "SELECT available_slots, subscription_start, subscription_expiry, signature
-                    FROM registration_receipts 
-                    WHERE tower_id = ?1 AND subscription_expiry = (SELECT MAX(subscription_expiry) 
-                        FROM registration_receipts 
-                        WHERE tower_id = ?1)",
-            )
-            .unwrap();
+        let mut query = "SELECT available_slots, subscription_start, subscription_expiry, signature FROM registration_receipts WHERE tower_id = ?1".to_string();
+
+        let mut params = vec![tower_id.to_vec()];
+
+        if let Some(start) = subscription_start {
+            query.push_str(" AND subscription_start >= ?2");
+            params.push(start.to_be_bytes().to_vec());
+        }
+
+        if let Some(expiry) = subscription_expiry {
+            query.push_str(" AND subscription_expiry <= ?3");
+            params.push(expiry.to_be_bytes().to_vec());
+        }
+
+        query.push_str(" ORDER BY subscription_expiry DESC LIMIT 1");
+
+        let mut stmt = self.connection.prepare(&query).unwrap();
+        let params: Vec<&dyn ToSql> = params.iter().map(|v| v as &dyn ToSql).collect();
```

Comment:
Notice that you've changed the behavior of this method.

If the bounds are not passed, the method should work exactly as it was before. Otherwise, it should have this new functionality. 

Also, why are you limiting the result? This can potentially return multiple items.

Reply:
Got it, I will keep exactly old method in case of  no optional parameters.
Actually I thought since old function was returning only latest receipt, new one should do same. I will remove the limit.

Reply:
There's no need for two methods tho, just build the SQL query based on what the input params are.